5_evaluate.R

```diff
@@ -47,109 +48,153 @@ p5 <- list(
              },
              format = 'file'),
   
+  # Get vector of evaluation sites, based on matched pred-obs
+  tar_target(p5_eval_sites,
+             p5_nldas_pred_obs %>% pull(site_id) %>% unique()),
+  
+  # Pull out lake depth for each evaluation site
+  tar_target(
+    p5_nldas_depths,
+    purrr::map_df(p5_eval_sites, function(site_id) {
+      site_nml <- p1_nldas_nml_list_subset[[site_id]]
+      tibble(
+        site_id = site_id,
+        lake_depth = site_nml$lake_depth
+      )
+    })
+  ),
+  
   # Prep matched preds for evaluation
   # Add pred_diff column (pred - obs)
   # Set up variables for which bias/accuracy will be calculated
   # Add fields for year, depth_class, doy, doy_bin, season, temp_bin
   tar_target(p5_nldas_pred_obs_eval,
-             prep_data_for_eval(p5_nldas_pred_obs, surface_cutoff_depth = 1, middle_cutoff_depth = 5,
-                                doy_bin_size = 5, temp_bin_size = 2)),
-  
-  # Filter to only surface pred-obs
-  tar_target(p5_nldas_pred_obs_eval_surface,
-             filter(p5_nldas_pred_obs_eval, depth_class=='surface')),
+             prep_data_for_eval(p5_nldas_pred_obs, p5_nldas_depths, surface_max_depth = 1, 
+                                bottom_depth_factor = 0.85, doy_bin_size = 5, temp_bin_size = 2)),
+  
+  # Filter evaluation pred-obs to surface and bottom predictions and set up for group mapping
+  tar_target(p5_nldas_pred_obs_eval_groups,
+             p5_nldas_pred_obs_eval %>%
+               filter(depth_class %in% c('surface','bottom')) %>%
+               group_by(depth_class) %>%
+               tar_group(),
+             iteration = "group"),
   
   ###### Assess model bias ######
   
   # Bias through time - year
-  tar_target(p5_nldas_surface_bias_year,
-             calc_bias(p5_nldas_pred_obs_eval_surface, grouping_var = 'year')),
+  tar_target(p5_nldas_bias_year,
+             calc_bias(p5_nldas_pred_obs_eval_groups, grouping_var = 'year',
+                       depth_class = unique(p5_nldas_pred_obs_eval_groups$depth_class)),
```

Comment:
Since this is getting a branch/grouped input, I'm assuming it only gets the part of the dataset that has "surface" or "bottom" for that class. If that is the case, `unique()` would be length == 1L. Do you want to put a simple `stopifnot(length(depth_class) == 1)` in your `calc_bias()` function? I'm assuming it will fail with the code 
```r
mutate(depth_class = depth_class, .before=1)
``` 
regardless if length(depth_class) != nrow(eval_pred_obs), but might want to use an error here to make the function's assumptions more clear.

Reply:
Good idea! I added that check to both the bias and rmse functions